scss/_patterns_segmented-control.scss

```diff
@@ -0,0 +1,60 @@
+@mixin vf-p-segmented-control {
+  %vf-button-group-borders {
+    border-bottom: 0;
+    border-left: 0;
+    border-right: 1px solid $color-mid-light;
+    border-top: 0;
+
+    &:last-child {
+      border-right: 0;
+    }
+  }
+
+  .p-segmented-control {
+    @extend %vf-has-round-corners;
+
+    border: 1px solid $color-mid-dark;
+    display: inline-flex;
+    margin-bottom: $spv-outer--small-scaleable;
+
+    // stylelint-disable selector-max-type
+    button,
+    [class*='p-button'] {
+      @extend %vf-button-group-borders;
+
+      border-radius: 0;
+      margin: 0;
+    }
+
+    .p-segmented-control__radio {
+      @extend %vf-button-group-borders;
+
+      margin-bottom: 0;
+      padding-top: 0;
+    }
+
+    [type='radio'] {
+      @extend %vf-hidden-tick-input;
+
+      &:checked + span {
+        background-color: $color-mid-x-light;
```

Comment:
The buttons are themed, this component shjould be able to leverage the theming. Have you tested with dark theme buttons? This colour would need to come from the themese from example.